exp/addons/internal/controllers/clusterresourceset_helpers.go

```diff
@@ -234,3 +234,29 @@ func normalizeData(resource *unstructured.Unstructured) ([][]byte, error) {
 
 	return dataList, nil
 }
+
+// ensureKubernetesServiceCreated ensures that the Service for Kubernetes API Server has been created.
+func ensureKubernetesServiceCreated(ctx context.Context, client client.Client) error {
+	err := client.Get(ctx, types.NamespacedName{
+		Namespace: metav1.NamespaceDefault,
+		Name:      "kubernetes",
+	}, &corev1.Service{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ensureKubeDNSServiceCreated ensures that the Service for kube-dns has been created.
+func ensureKubeDNSServiceCreated(ctx context.Context, client client.Client) error {
+	err := client.Get(ctx, types.NamespacedName{
+		Namespace: metav1.NamespaceSystem,
+		Name:      "kube-dns",
```

Comment:
AFAIK this is the name used by kubeadm, probably by some other setup scripts, but it is not an official convention/standard defined somewhere.
If this is true, we can't back in into our code this assumption unless we find a way to make this generic

Reply:
agree, we can remove the wait logic for "kube-dns".

Reply:
done